File: guess_number.scala
#! /usr/bin/env scala
!#

import scala.util.Random

println("Guess-a-number game!")
println("I am thinking of a number between 1 and 100")
print("What's your guess? ")

val target = Random.nextInt(100) + 1
var current = 0

while (current != target) {

  current = Console.readInt
  if (current < target)
    print("Too low! Guess again: ")
  else if (current > target)
    print("Too high! Guess again: ")

}

println("Correct! You won!")

/** TAIL INFO:
 * Name: Guess Number
 * Language: Scala
 * State: Done
 *
 * Play guess-a-number game
 *
 *
 * Example: ./guess_number.scala
*/

File: guess_number.java

import java.util.Random;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class guess_number {
  public static void main(String[] args) {
    
    System.out.println("Guess-a-number game!");
    System.out.println("I am thinking of a number between 1 and 100");
    System.out.print("What's your guess? ");
    System.out.flush();
    
    int target = new Random().nextInt(100) + 1;
    int current = 0;
    BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));

    while (current != target) {
      
      try {
        current = Integer.parseInt(stdin.readLine());
      } catch (IOException e) {}
      
      if (current < target)
        System.out.print("Too Low! Try again: ");
      else if (current > target)
        System.out.print("Too High! Try again: ");
      else {
        System.out.println("Correct! You won!");
        return;
      }
      System.out.flush();
    }
  }
}

/* TAIL INFO:
Name: Guess Number
Language: Java
Compile: javac guess_number.java
State: Done

Play guess-a-number game


Example: java guess_number
*/
